src/appleseed/renderer/modeling/camera/camera.cpp

```diff
@@ -64,6 +64,40 @@ namespace
     const UniqueID g_class_uid = new_guid();
 }
 
+struct Camera::Impl
+{
+    //
+    // Linear shutter parameters.
+    //
+
+    float                       m_shutter_curve_linear_open_multiplier;
+    float                       m_shutter_curve_linear_fully_opened_multiplier;
+    float                       m_shutter_curve_linear_close_multiplier;
+
+    //
+    // Bezier shutter curve parameters.
+    //
+
+    // Control points are represented in form [t0 s0 t1 s1 t2 s2 t3 s3]
+    // where t stands for time and s stands for "shutter openness".
+    typedef Vector<float, 8> ShutterCurveControlPoints;
+
+    bool                        m_use_bezier_shutter_curve;
+    ShutterCurveControlPoints   m_shutter_curve_bezier_control_points_normalized;
+
+    // Normalization factor to make the area under the curve equal to 1.
+    float                       m_shutter_curve_bezier_normalization_factor;
+
+    // Polynomials of open/close CDFs.
+    Vector<float, 7>            m_shutter_curve_bezier_open_cdf;
+    Vector<float, 7>            m_shutter_curve_bezier_close_cdf;
+
+    Impl()
+      : m_use_bezier_shutter_curve(false)
```

Comment:
This probably should go to on_render_begin()

Reply:
Good point!